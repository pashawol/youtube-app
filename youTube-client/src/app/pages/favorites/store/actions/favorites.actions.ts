import { Item } from "@app/shared/models/response.model"
import { createAction, props } from "@ngrx/store"

export const loadFavorites = createAction("[Favorites] Load Favorites")
export const loadFavoritesSuccess = createAction("[Favorites] Load Favorites Success", props<{ favorites: string[] }>())

export const toggleFavorite = createAction("[Favorite] Toggle Favorite", props<{ id: string }>())

export const loadFavoritesData = createAction("[Favorites] Load Favorites Data")
export const loadFavoritesDataSuccess = createAction(
    "[Favorites] Load Favorites Data Success",
    props<{ favoritesData: Item[] }>()
)
export const loadFavoritesDataFailure = createAction(
    "[Favorites] Load Favorites Data Failure",
    props<{ error: string | null }>()
)
