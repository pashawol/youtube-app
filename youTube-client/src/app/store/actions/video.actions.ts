import { createAction, props } from "@ngrx/store"
import { Item } from "@shared/models/response.model"

import { VideoState } from "../videos.state"

export const getVideoById = createAction("[Video] Get Video By Id", props<{ id: string }>())
export const getVideoByIdSuccess = createAction("[Video] Get Video By Id Success", props<{ video: Item }>())
export const getVideoByIdFailure = createAction("[Video] Get Video By Id Failure", props<{ error: string | null }>())

export const loadVideos = createAction("[Video] Load Videos")
export const loadVideosSuccess = createAction("[Video] Load Videos Success", props<{ videos: Item[] }>())
export const loadVideosFailure = createAction("[Video] Load Videos Failure", props<{ error: string | null }>())

export const createVideo = createAction("[Video] Create Video", props<{ video: Item }>())
export const createVideoFailure = createAction("[Video] Create Video Failure", props<{ error: string | null }>())

export const deleteVideo = createAction("[Video] Delete Video", props<{ id: string }>())
