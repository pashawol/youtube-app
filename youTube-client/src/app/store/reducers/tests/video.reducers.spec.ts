import { Item } from "@app/shared/models/response.model"

import * as VideoActions from "../../actions/video.actions"
import { initialState } from "../../videos.state"
import { videoReducer } from "../video.reducer"

const newVideo: Item = {
    kind: "kind",
    etag: "etag",
    id: "id",
    snippet: {
        publishedAt: "publishedAt",
        channelId: "channelId",
        title: "title",
        description: "description",
        thumbnails: {
            default: {
                url: "url",
                width: 0,
                height: 0
            },
            medium: {
                url: "url",
                width: 0,
                height: 0
            },
            high: {
                url: "url",
                width: 0,
                height: 0
            }
        },
        channelTitle: "channelTitle",
        tags: ["tags"],
        categoryId: "categoryId",
        liveBroadcastContent: "liveBroadcastContent",
        localized: {
            title: "title",
            description: "description"
        },
        defaultAudioLanguage: "defaultAudioLanguage",
        defaultLanguage: "defaultLanguage"
    },
    statistics: {
        viewCount: "viewCount",
        likeCount: "likeCount",
        dislikeCount: "dislikeCount",
        favoriteCount: "favoriteCount",
        commentCount: "commentCount"
    }
}
describe("videoReducer", () => {
    it("should return the initial state", () => {
        const state = videoReducer(undefined, { type: "@@INIT" })
        expect(state).toMatchSnapshot()
    })

    it("should handle createVideo action", () => {
        const action = VideoActions.createVideo({ video: newVideo })
        const state = videoReducer(initialState, action)

        expect(state.localVideos).toMatchSnapshot()
    })

    it("should handle loadVideosSuccess action", () => {
        const action = VideoActions.loadVideosSuccess({ videos: [newVideo] })
        const state = videoReducer(initialState, action)

        expect(state.videos).toMatchSnapshot()
    })

    it("should handle getVideoByIdSuccess action", () => {
        const action = VideoActions.getVideoByIdSuccess({ video: newVideo })
        const state = videoReducer(initialState, action)

        expect(state.video).toMatchSnapshot()
    })
})
