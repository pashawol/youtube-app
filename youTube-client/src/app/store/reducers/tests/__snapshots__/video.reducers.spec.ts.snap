// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`videoReducer should handle createVideo action 1`] = `
[
  {
    "etag": "",
    "id": "local-obUsyRdkHaM",
    "kind": "",
    "snippet": {
      "categoryId": "local",
      "channelId": "",
      "channelTitle": "",
      "defaultAudioLanguage": "",
      "defaultLanguage": "",
      "description": "video",
      "liveBroadcastContent": "",
      "localized": {
        "description": "video",
        "title": "New video",
      },
      "publishedAt": "08-10-2024",
      "tags": [
        "teg",
        "tag 1",
        "teg 3",
      ],
      "thumbnails": {
        "default": {
          "height": 0,
          "url": "https://media.licdn.com/dms/image/D4E22AQFcylssXyxIAw/feedshare-shrink_800/0/1723378128666?e=1726099200&v=beta&t=MvHvUDQ6kE6SAISk8NooNnRt2Vqw-2KTkbmXZhr9O9o",
          "width": 0,
        },
        "high": {
          "height": 0,
          "url": "https://media.licdn.com/dms/image/D4E22AQFcylssXyxIAw/feedshare-shrink_800/0/1723378128666?e=1726099200&v=beta&t=MvHvUDQ6kE6SAISk8NooNnRt2Vqw-2KTkbmXZhr9O9o",
          "width": 0,
        },
        "medium": {
          "height": 0,
          "url": "https://media.licdn.com/dms/image/D4E22AQFcylssXyxIAw/feedshare-shrink_800/0/1723378128666?e=1726099200&v=beta&t=MvHvUDQ6kE6SAISk8NooNnRt2Vqw-2KTkbmXZhr9O9o",
          "width": 0,
        },
      },
      "title": "New video",
    },
    "statistics": {
      "commentCount": "0",
      "dislikeCount": "0",
      "favoriteCount": "0",
      "likeCount": "0",
      "viewCount": "0",
    },
  },
  {
    "etag": "",
    "id": "local-bCBHoHLlMFs",
    "kind": "",
    "snippet": {
      "categoryId": "local",
      "channelId": "",
      "channelTitle": "",
      "defaultAudioLanguage": "",
      "defaultLanguage": "",
      "description": "video 2",
      "liveBroadcastContent": "",
      "localized": {
        "description": "video 2",
        "title": "New video 2",
      },
      "publishedAt": "08-10-2024",
      "tags": [
        "teg",
        "tag 1",
        "teg 3",
      ],
      "thumbnails": {
        "default": {
          "height": 0,
          "url": "https://media.licdn.com/dms/image/D4E22AQHXYQf4oyYg1Q/feedshare-shrink_800/0/1723389466302?e=1726099200&v=beta&t=-Ml16Qq4KHKI_PdrFO2GdFWa6-SCIkTpgxigCdOfWUg",
          "width": 0,
        },
        "high": {
          "height": 0,
          "url": "https://media.licdn.com/dms/image/D4E22AQHXYQf4oyYg1Q/feedshare-shrink_800/0/1723389466302?e=1726099200&v=beta&t=-Ml16Qq4KHKI_PdrFO2GdFWa6-SCIkTpgxigCdOfWUg",
          "width": 0,
        },
        "medium": {
          "height": 0,
          "url": "https://media.licdn.com/dms/image/D4E22AQHXYQf4oyYg1Q/feedshare-shrink_800/0/1723389466302?e=1726099200&v=beta&t=-Ml16Qq4KHKI_PdrFO2GdFWa6-SCIkTpgxigCdOfWUg",
          "width": 0,
        },
      },
      "title": "New video 2",
    },
    "statistics": {
      "commentCount": "0",
      "dislikeCount": "0",
      "favoriteCount": "0",
      "likeCount": "0",
      "viewCount": "0",
    },
  },
  {
    "etag": "etag",
    "id": "id",
    "kind": "kind",
    "snippet": {
      "categoryId": "categoryId",
      "channelId": "channelId",
      "channelTitle": "channelTitle",
      "defaultAudioLanguage": "defaultAudioLanguage",
      "defaultLanguage": "defaultLanguage",
      "description": "description",
      "liveBroadcastContent": "liveBroadcastContent",
      "localized": {
        "description": "description",
        "title": "title",
      },
      "publishedAt": "publishedAt",
      "tags": [
        "tags",
      ],
      "thumbnails": {
        "default": {
          "height": 0,
          "url": "url",
          "width": 0,
        },
        "high": {
          "height": 0,
          "url": "url",
          "width": 0,
        },
        "medium": {
          "height": 0,
          "url": "url",
          "width": 0,
        },
      },
      "title": "title",
    },
    "statistics": {
      "commentCount": "commentCount",
      "dislikeCount": "dislikeCount",
      "favoriteCount": "favoriteCount",
      "likeCount": "likeCount",
      "viewCount": "viewCount",
    },
  },
]
`;

exports[`videoReducer should handle getVideoByIdSuccess action 1`] = `
{
  "etag": "etag",
  "id": "id",
  "kind": "kind",
  "snippet": {
    "categoryId": "categoryId",
    "channelId": "channelId",
    "channelTitle": "channelTitle",
    "defaultAudioLanguage": "defaultAudioLanguage",
    "defaultLanguage": "defaultLanguage",
    "description": "description",
    "liveBroadcastContent": "liveBroadcastContent",
    "localized": {
      "description": "description",
      "title": "title",
    },
    "publishedAt": "publishedAt",
    "tags": [
      "tags",
    ],
    "thumbnails": {
      "default": {
        "height": 0,
        "url": "url",
        "width": 0,
      },
      "high": {
        "height": 0,
        "url": "url",
        "width": 0,
      },
      "medium": {
        "height": 0,
        "url": "url",
        "width": 0,
      },
    },
    "title": "title",
  },
  "statistics": {
    "commentCount": "commentCount",
    "dislikeCount": "dislikeCount",
    "favoriteCount": "favoriteCount",
    "likeCount": "likeCount",
    "viewCount": "viewCount",
  },
}
`;

exports[`videoReducer should handle loadVideosSuccess action 1`] = `
[
  {
    "etag": "etag",
    "id": "id",
    "kind": "kind",
    "snippet": {
      "categoryId": "categoryId",
      "channelId": "channelId",
      "channelTitle": "channelTitle",
      "defaultAudioLanguage": "defaultAudioLanguage",
      "defaultLanguage": "defaultLanguage",
      "description": "description",
      "liveBroadcastContent": "liveBroadcastContent",
      "localized": {
        "description": "description",
        "title": "title",
      },
      "publishedAt": "publishedAt",
      "tags": [
        "tags",
      ],
      "thumbnails": {
        "default": {
          "height": 0,
          "url": "url",
          "width": 0,
        },
        "high": {
          "height": 0,
          "url": "url",
          "width": 0,
        },
        "medium": {
          "height": 0,
          "url": "url",
          "width": 0,
        },
      },
      "title": "title",
    },
    "statistics": {
      "commentCount": "commentCount",
      "dislikeCount": "dislikeCount",
      "favoriteCount": "favoriteCount",
      "likeCount": "likeCount",
      "viewCount": "viewCount",
    },
  },
]
`;

exports[`videoReducer should return the initial state 1`] = `
{
  "currentPagination": null,
  "error": null,
  "localVideos": [
    {
      "etag": "",
      "id": "local-obUsyRdkHaM",
      "kind": "",
      "snippet": {
        "categoryId": "local",
        "channelId": "",
        "channelTitle": "",
        "defaultAudioLanguage": "",
        "defaultLanguage": "",
        "description": "video",
        "liveBroadcastContent": "",
        "localized": {
          "description": "video",
          "title": "New video",
        },
        "publishedAt": "08-10-2024",
        "tags": [
          "teg",
          "tag 1",
          "teg 3",
        ],
        "thumbnails": {
          "default": {
            "height": 0,
            "url": "https://media.licdn.com/dms/image/D4E22AQFcylssXyxIAw/feedshare-shrink_800/0/1723378128666?e=1726099200&v=beta&t=MvHvUDQ6kE6SAISk8NooNnRt2Vqw-2KTkbmXZhr9O9o",
            "width": 0,
          },
          "high": {
            "height": 0,
            "url": "https://media.licdn.com/dms/image/D4E22AQFcylssXyxIAw/feedshare-shrink_800/0/1723378128666?e=1726099200&v=beta&t=MvHvUDQ6kE6SAISk8NooNnRt2Vqw-2KTkbmXZhr9O9o",
            "width": 0,
          },
          "medium": {
            "height": 0,
            "url": "https://media.licdn.com/dms/image/D4E22AQFcylssXyxIAw/feedshare-shrink_800/0/1723378128666?e=1726099200&v=beta&t=MvHvUDQ6kE6SAISk8NooNnRt2Vqw-2KTkbmXZhr9O9o",
            "width": 0,
          },
        },
        "title": "New video",
      },
      "statistics": {
        "commentCount": "0",
        "dislikeCount": "0",
        "favoriteCount": "0",
        "likeCount": "0",
        "viewCount": "0",
      },
    },
    {
      "etag": "",
      "id": "local-bCBHoHLlMFs",
      "kind": "",
      "snippet": {
        "categoryId": "local",
        "channelId": "",
        "channelTitle": "",
        "defaultAudioLanguage": "",
        "defaultLanguage": "",
        "description": "video 2",
        "liveBroadcastContent": "",
        "localized": {
          "description": "video 2",
          "title": "New video 2",
        },
        "publishedAt": "08-10-2024",
        "tags": [
          "teg",
          "tag 1",
          "teg 3",
        ],
        "thumbnails": {
          "default": {
            "height": 0,
            "url": "https://media.licdn.com/dms/image/D4E22AQHXYQf4oyYg1Q/feedshare-shrink_800/0/1723389466302?e=1726099200&v=beta&t=-Ml16Qq4KHKI_PdrFO2GdFWa6-SCIkTpgxigCdOfWUg",
            "width": 0,
          },
          "high": {
            "height": 0,
            "url": "https://media.licdn.com/dms/image/D4E22AQHXYQf4oyYg1Q/feedshare-shrink_800/0/1723389466302?e=1726099200&v=beta&t=-Ml16Qq4KHKI_PdrFO2GdFWa6-SCIkTpgxigCdOfWUg",
            "width": 0,
          },
          "medium": {
            "height": 0,
            "url": "https://media.licdn.com/dms/image/D4E22AQHXYQf4oyYg1Q/feedshare-shrink_800/0/1723389466302?e=1726099200&v=beta&t=-Ml16Qq4KHKI_PdrFO2GdFWa6-SCIkTpgxigCdOfWUg",
            "width": 0,
          },
        },
        "title": "New video 2",
      },
      "statistics": {
        "commentCount": "0",
        "dislikeCount": "0",
        "favoriteCount": "0",
        "likeCount": "0",
        "viewCount": "0",
      },
    },
  ],
  "video": null,
  "videos": null,
}
`;
